Coordinate.h

```diff
@@ -1,13 +1,21 @@
 #pragma once
 
 #include "Direction.h"
-
+#define HASH_CONST 1000
 class Coordinate {
   protected:
     Coordinate();
     Coordinate(int x, int y);
+    Coordinate(Direction d);
+    int hash() const;
 
   public:
+    Coordinate operator*(int a);
+    Coordinate operator-(const Coordinate &a);
+    Coordinate operator+(const Coordinate &a);
+    bool operator<(const Coordinate &a) const;
```

Comment:
The < is used for the std::set to store the unvisited vertices in the algorithm. The other three, along with the constructor that accepts a Direction, help for the implementation of the overloads for the NodeCoordinate operator+(Direction d) and operator+(DirectionVector directvect)